packages/backend/src/models/GroupsModel.ts

```diff
@@ -221,4 +222,17 @@ export class GroupsModel {
             organizationUuid: updatedGroup.organization_uuid,
         };
     }
+
+    async addUserToGroupIfItExists(userUuid: string, groups: string[]) {
+        const existingGroupsUuids = await this.database('groups')
+            .whereIn('name', groups)
+            .select('group_uuid', 'name', 'organization_id');
+
+        existingGroupsUuids.forEach(async (group) => {
```

Comment:
You could do this with a single query instead of reusing an existing model method, which is going to execute queries in a loop. If somebody had a lot of groups this could get expensive

Reply:
sounds good to me! looking into it now 👍

Reply:
done here: https://github.com/lightdash/lightdash/pull/8508/commits/25d2f63b6f3cd71cc3b75714d97a982b4b0b9990